File: python_programmers_school/level 1/없는 숫자 더하기.py
def solution(numbers):
    a = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
    return sum(list(a - set(numbers)))

# 간단한 풀이
# def solution(numbers):
#     return 45 - sum(numbers)     --> 0부터 9까지의 수를 모두 더한 값인 45에서 numbers 리스트의 수를 모두 더한 값을 빼면 바로 답이 나옴.

# 람다 함수 사용한 간단한 풀이
# solution = lambda x: sum(range(10)) - sum(x)

File: javascript_programmers_school/level 1/없는 숫자 더하기.js
function solution(numbers) {
  const pick = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9].filter(
    (num) => numbers.includes(num) === false
  );
  return pick.reduce((acc, cur) => acc + cur);
}

// 다른 사람의 풀이
// 배열에 없는 숫자의 합을 구하는 것이므로 0부터 9까지 숫자의 합에서 배열에 있는 모든 숫자를 빼는 방식으로 작성하면 코드가 더 깔끔해진다.
function solution(numbers) {
  return 45 - numbers.reduce((cur, acc) => cur + acc, 0);
}
